internal/service/ch/queries.go

```diff
@@ -375,21 +375,28 @@ func getAggQuery(aggArgs *model.AggregatedSignalArgs) (string, []any, error) {
 	}
 	valueTable := fmt.Sprintf("VALUES('%s', %s) as %s ON %s.%s = %s.%s", valueTableDef, strings.Join(valuesArgs, ", "), aggTableName, vss.TableName, vss.NameCol, aggTableName, vss.NameCol)
 
-	floatFilters := []qm.QueryMod{
-		// Make sure non-float rows can still get returned.
+	// This NEQ ensures that non-float rows get returned.
+	// TODO(elffjs): When we have filtering for more field types, this
+	// will get more complex.
+	perSignalFilters := []qm.QueryMod{
 		qmhelper.Where(signalTypeCol, qmhelper.NEQ, FloatType),
 	}
 
-	for i, agg := range aggArgs.FloatArgs {
-		// TODO(elffjs): Some duplication here. Also a bit wasteful if
-		// there are no filters at all.
-		fieldFilters := []qm.QueryMod{
-			qmhelper.Where(signalTypeCol, qmhelper.EQ, FloatType),
-			qmhelper.Where(signalIndexCol, qmhelper.EQ, i),
+	if len(aggArgs.FloatArgs) != 0 {
```

Comment:
I don't like this guard, but if `innerFloatFilters` is empty then the query will look like `… AND (field_type == 1 OR ())` and ClickHouse will reject it.

Reply:
yeah other option is to create the list with `1 == 1`  and that feels just as bad